Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/ProhibitDynamicKeywordAnalyzer.cs

```diff
@@ -9,49 +10,94 @@
 namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
 {
 	[DiagnosticAnalyzer(LanguageNames.CSharp)]
-	public class ProhibitDynamicKeywordAnalyzer : SingleDiagnosticAnalyzer<IdentifierNameSyntax, ProhibitDynamicKeywordSyntaxNodeAction>
```

Comment:
It's hard to follow the diffs. What's happening in this class?

Reply:
I transferred it from Trivia based into declaration based. This saved a ton of `SemanticModel` lookups, which are now only done when we know 90% sure its a `dynamic` type. Previsouly this analyzer dis a `SemanticModel` lookup for _every_ `Identifier` in the entire code.